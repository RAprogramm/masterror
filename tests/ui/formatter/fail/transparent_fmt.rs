// SPDX-FileCopyrightText: 2025 RAprogramm <andrey.rozanov.vl@gmail.com>
//
// SPDX-License-Identifier: MIT

use masterror::Error;

fn format_unit(_f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
    Ok(())
}

#[derive(Debug, Error)]
#[error(transparent, fmt = crate::format_unit)]
struct TransparentFormatterWrapper(#[from] std::io::Error);

fn main() {}
