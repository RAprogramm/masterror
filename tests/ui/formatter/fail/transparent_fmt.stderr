error: format arguments are not supported with #[error(transparent)]
  --> tests/ui/formatter/fail/transparent_fmt.rs:12:20
   |
12 | #[error(transparent, fmt = crate::format_unit)]
   |                    ^
